feat: tela de remessas e um monte de coisa

Cauatn opened the pull request.

Cauatn committed:
feat: tank managment impl screen

Cauatn committed:
feat: data mock table

jvictordev1 committed:
Merge pull request #4 from scanai-startup/dev

release/0.0.0

Cauatn committed:
feat: update authentication flow and form schema

- Refactored SignInForm to use new fields 'matricula' and 'senha'.
- Integrated AuthService for login handling and added loading state.
- Updated signinForm schema to reflect new field names.
- Added LogoutButton to the AppSidebar for user logout functionality.
- Exported AuthService and LogoutButton from the auth module.

Cauatn committed:
Merge branch 'cauatn/tanks-managment' of https://github.com/scanai-startup/scanai-web

Cauatn committed:
feat: implement remessa management features

- Added NovaRemessaSidebar component for creating new remessas.
- Updated RemessasPage to include new sidebar and enhanced UI with cards and tabs.
- Refactored RemessaStatsComponent to improve display of statistics.
- Introduced detailed view for remessas with a new VerMaisButton in the RemessaTable.
- Enhanced RemessaService with create, update, and delete functionalities for remessas.
- Updated data fetching logic to handle backend errors gracefully.

Cauatn committed:
Merge branch 'feat/remessa' of https://github.com/scanai-startup/scanai-web into feat/remessas

Cauatn committed:
refactor: reorganizar estrutura de arquivos para pasta /app

- Mover páginas de remessas e tanques para /app/app/dashboard/
- Atualizar rotas no sidebar para novos caminhos
- Remover arquivos antigos da estrutura /dashboard

Cauatn committed:
feat: implementar sidesheet para nova remessa

- Criar RemessaDetailsSidesheet seguindo padrão da gestão de tanques
- Suporte para modo create e edit no mesmo componente
- Formulário completo com todas as seções necessárias
- Integração com RemessaService para CRUD operations

Cauatn committed:
refactor: implementar hook customizado e organizar componentes

- Criar useRemessas hook para gerenciar estado e lógica
- Separar componentes de cards em arquivos individuais
- Refatorar página de remessas para usar hook e componentes organizados
- Seguir diretrizes do React para evitar useEffect desnecessário
- Melhorar legibilidade e manutenibilidade do código

Cauatn committed:
refactor: separar tabela de remessas em arquivos organizados

- Criar remessa-columns.tsx com definições das colunas
- Criar remessa-table.tsx com componente da tabela
- Separar responsabilidades entre colunas e lógica da tabela
- Melhorar organização e reutilização de componentes
- Atualizar exportações no index.ts

Cauatn committed:
chore: atualizar configurações e tipos

- Atualizar constantes de roles
- Melhorar configurações de autenticação
- Ajustar configurações do TypeScript
- Backup do middleware antigo

Cauatn committed:
chore: remove old middleware backup file

Cauatn committed:
refactor: update SignInForm component for improved authentication flow

- Replace AuthService with signIn function for API calls
- Integrate useApiCall hook to manage loading state
- Enhance error handling with user-friendly messages
- Update form submission logic and loading indicators

jvictordev1 commented:
Acho melhor deixar os selects de 'Informações do Vinho' da largura do container. E tb adicionar um padding nesse sidesheet (rola a mesma coisa no sidesheet de editar remessa).

<img width="597" height="1007" alt="image" src="https://github.com/user-attachments/assets/74919e9e-8d71-489f-867d-da1261687408" />

jvictordev1 commented:
Os campos de caixas, peso, sanidade e ações tão desalinhados. No de ações acho melhor usar só icones, igual fiz na tela de gestão de tanques

<img width="1555" height="635" alt="image" src="https://github.com/user-attachments/assets/0efc0abc-0896-44a1-8309-3263beca99e7" />

jvictordev1 commented on src/app/app/dashboard/remessas/page.tsx:
```diff
@@ -0,0 +1,90 @@
+} from '@/features/remessa';
+
+export default function RemessasPage() {
+	const { remessas, stats, loading, error, refreshRemessas } = useRemessas();
```
Eu tinha criado um `useApiCall` na branch de gerenciamento de tanques, acho q dava pra reutilizar aqui também. Lá já fica responsável por retornar o data do request, gerenciar o loading e dar throw no error (ai daria pra ver se a gnt mantem esse comportamento do throw ou dá return um estado de erro)

jvictordev1 commented on src/app/app/dashboard/tanks/page.tsx:
Eu já fiz essa tela na outra banch doido

jvictordev1 commented on src/features/remessa/components/cards/RemessaStatusCard.tsx:
```diff
@@ -0,0 +1,64 @@
+
+export function RemessaStatusCard({ stats }: RemessaStatusCardProps) {
+	const total = stats.remessasAtivas + stats.remessasInativas;
+	const getPercent = (value: number) => (value / total) * 100;
```
Joga essa parada pra utils, q a gnt ja reaproveita em outros cantos

jvictordev1 commented on src/features/remessa/components/remessa-stats.tsx:
Esse componente aq n ta sendo usado mais

jvictordev1 commented on src/features/remessa/components/tables/remessa-columns.tsx:
```diff
@@ -0,0 +1,251 @@
+		accessorKey: 'casta',
+		header: 'Casta',
+		cell: ({ row }) => (
+			<div className='font-medium'>{row.getValue('casta')}</div>
```
Acho q fica melhor componentizar essas divs com `font-medium`, pra n ficar repetindo toda hora o código quando quiser aplicar só esse estilo. Além disso padroniza pra o resto das outras tabelas da aplicação

jvictordev1 commented on src/features/remessa/components/tables/remessa-table.tsx:
```diff
@@ -0,0 +1,127 @@
+import { remessaColumns } from './remessa-columns';
+
+export function RemessaTable({ data }: { data: Remessa[] }) {
+	const [sorting, setSorting] = React.useState<SortingState>([]);
```
Como nenhuma coluna ta com `enableSorting` daria pra tirar isso, pq n ta dando pra dar sort

jvictordev1 commented on src/features/remessa/hooks/useRemessas.ts:
Não seria melhor usar o `useApiCall` pra lidar com o request? Pq tipo, fica repetitivo fazer vários states lidarem com o request, loading e error. Acho que a gnt podia centralizar tudo lá.

jvictordev1 commented on src/features/remessa/hooks/useRemessas.ts:
```diff
@@ -0,0 +1,82 @@
+	const [loading, setLoading] = useState(true);
+	const [error, setError] = useState<string | null>(null);
+
+	const calculateStats = (data: Remessa[]): RemessaStats => {
```
Bom falar c os caras p eles criarem um endpoint p retornar isso p gente, p n ficar calculando no front

Cauatn replied:
tem que falar, mas do jeito que ta so outra vida pra fazer

jvictordev1 commented on src/features/remessa/services/remessa-service.ts:
```diff
@@ -0,0 +1,187 @@
+export class RemessaService {
+	static async getAllRemessas(): Promise<Remessa[]> {
+		try {
+			const response = await fetch(`${API_BASE_URL}/uva/getAll`, {
```
O `apiService` encapsula toda essa lógica inicial de setar os cabeçalhos e criar a url. Acho que da p trocar o uso do fetch nesse service pelo uso do `apiService`

jvictordev1 commented on src/features/tank/components/Tables/Companies/columns.tsx:
Essa parte ta na minha branch já

jvictordev1 commented on src/features/tank/components/Tables/Companies/data-table.tsx:
Tb ta na minha branch esse componente

jvictordev1 commented on src/features/auth/services/auth-service.ts:
Essa classe foi criada mas o método não ta sendo usado no login. Joga o método `signIn` do arquivo `authService.ts` pra cá e atualiza o import lá no `SignInForm`.

Outra coisa tb, a classe ta levando em consideração que a gnt ta salvando token em localstorage, mas ta em cookie

jvictordev1 commented on src/features/auth/services/auth-service.ts:
```diff
@@ -0,0 +1,62 @@
+const API_BASE_URL = 'http://localhost:8080';
+
+// Interface para dados de login
+interface LoginData {
```
Essa interface n ta sendo usada, e pro login a gnt pode usar o `SigninFormType` do zod

Cauatn committed:
fix: corrige loop infinito no useApiCall e useRemessas

- Memoiza função action no useApiCall usando useCallback
- Estabiliza referências de função para evitar re-renders desnecessários
- Corrige dependências do useEffect em useRemessas

Cauatn committed:
refactor: remove sortingState desnecessário da RemessaTable

- Remove SortingState já que nenhuma coluna tem enableSorting
- Remove imports e lógica relacionados a sorting não utilizados

Cauatn committed:
refactor: consolida autenticação em auth-service

- Move método signIn de authService.ts para auth-service.ts
- Remove referências a localStorage (token agora em cookie httpOnly)
- Remove interface LoginData não utilizada, usa SigninFormType do zod
- Adiciona função signOut para limpar cookie
- Remove arquivo authService.ts duplicado
- Atualiza LogoutButton para usar server action signOut

Cauatn committed:
refactor: converte RemessaService para usar apiService

- Converte métodos para server actions usando 'use server'
- Substitui fetch direto por apiService que encapsula headers e URL
- Remove dependência de AuthService.getToken (token agora em cookie)
- Mantém compatibilidade através de objeto RemessaService exportado

Cauatn committed:
refactor: atualiza página de remessas após refatoração do hook

- Remove tratamento de error state (agora gerenciado internamente)
- Simplifica lógica de loading usando useRemessas refatorado

Cauatn committed:
refactor: melhora UI da tabela de remessas

- Substitui botões de ações por ícones (Eye e Edit)
- Alinha colunas numéricas e ações à direita
- Mantém consistência com tela de gestão de tanques

Cauatn committed:
refactor: melhora sidesheet de remessas

- Adiciona padding horizontal no conteúdo do formulário
- Ajusta selects de 'Informações do Vinho' para largura completa
- Substitui botão de editar por ícone apenas
- Corrige tipos do formulário para corresponder ao tipo Remessa
- Adiciona campos obrigatórios fkfuncionario e fkmostro no create
